### ISSUE TITLE
Include libs and utilities for Portmaster

### ISSUE THREAD

**Issue by dxdatabase** [cizia64/CrossMix-OS#67]
As discussed in "Port Master apps not working" issue, a lot of libs are missing for Portmaster. The repository [Kiwi Addon Collection for tsp](https://github.com/dokoma/kiwi/tree/main) had compiled useful libs, it'll be very nice to include it to the next release.

He also compiled bash, useful too for portmaster's scripts (Busybox's Ash have limitations, preventing some ports to launch).

**Comment by sirchia** [comment]
Bash for additional Portmaster script compatibility might not be needed anymore as my PR has been merged:
https://github.com/PortsMaster/PortMaster-New/pull/466

Unless there are also other types of  incompatibilities. In that case please let me know what they are, perhaps they can be eliminated as well.

**Comment by dxdatabase** [comment]
Array is another problem with busybox. For example in Super Mario War.sh,   `for j in "${settings[@]}"` will not work. A few ports uses arrays.

But the main problem for running Portmaster's ports on Trimui is missing libs.

**Comment by cizia64** [comment]
We can run bash on the TSP (I think PortMaster is using it).
I've start a [compatibility list for PostMaster](https://docs.google.com/spreadsheets/d/1oCGa4lMDsZD4V7A559urMNcVzCZluptY8lyFmE45w5o/edit?gid=389157348#gid=389157348) on the TSP few times ago, the objective was to find solutions and complete the missings libs. It's a big work but with some help we can fix many stuffs by adding missing libs. However it must be organized and documented : starting by referencing the missing libs for the most famous ports is a good start.

**Comment by dxdatabase** [comment]
It's often the same missing libs : libpulse, libvorbis, libn, libpng16, libmpg123, libsndfile, libFLAC... It could be easily included and will solve a lot of Portmaster's problems. And libraries are small, don't mater if they are unused.

**Comment by ogregorio** [comment]
> We can run bash on the TSP (I think PortMaster is using it).
> I've start a [compatibility list for PostMaster](https://docs.google.com/spreadsheets/d/1oCGa4lMDsZD4V7A559urMNcVzCZluptY8lyFmE45w5o/edit?gid=389157348#gid=389157348) on the TSP few times ago, the objective was to find solutions and complete the missings libs. It's a big work but with some help we can fix many stuffs by adding missing libs. However it must be organized and documented : starting by referencing the missing libs for the most famous ports is a good start. 

Wow, very nice initiative!

I wonder if I can import your list into handhelddb.com

**Comment by cizia64** [comment]
> It's often the same missing libs : libpulse, libvorbis, libn, libpng16, libmpg123, libsndfile, libFLAC... It could be easily included and will solve a lot of Portmaster's problems. And libraries are small, don't mater if they are unused.

Where are located these libs ? [Here](https://github.com/dokoma/kiwi/tree/main/Emus/EASYRPGX/libs) ? 

> https://github.com/dokoma/kiwi/tree/main/Emus/EASYRPGX/libs

I was expecting more help on this from the community 😅 but yes feel free to use it as you want ;)

**Comment by tyusha0** [comment]
> Мы можем запустить bash на TSP (Я думаю, что PortMaster использует его). Я начал [список совместимости для PostMaster](https://docs.google.com/spreadsheets/d/1oCGa4lMDsZD4V7A559urMNcVzCZluptY8lyFmE45w5o/edit?gid=389157348#gid=389157348) на TSP несколько раз назад цель состояла в том, чтобы найти решения и завершить пропуски. Это большая работа, но с некоторой помощью мы можем исправить многие вещи, добавив недостающие либы. Однако он должен быть организован и задокументирован: начать с ссылки на недостающие libs для самых известных портов - хорошее начало.

Cave Story (Evo) missing libs: libpng16.so.16, libSDL2_mixer <= 2.0.4 im using sdl2_mixer 2.8.0-1 from https://archlinuxarm.org/packages/aarch64/sdl2_mixer

**Comment by dxdatabase** [comment]
> > It's often the same missing libs : libpulse, libvorbis, libn, libpng16, libmpg123, libsndfile, libFLAC... It could be easily included and will solve a lot of Portmaster's problems. And libraries are small, don't mater if they are unused.
> 
> Where are located these libs ? [Here](https://github.com/dokoma/kiwi/tree/main/Emus/EASYRPGX/libs) ?
> 
> > https://github.com/dokoma/kiwi/tree/main/Emus/EASYRPGX/libs
> 

Yes, and [here](https://github.com/dokoma/kiwi/tree/main/Emus/LOVE/libs), [here](https://github.com/dokoma/kiwi/tree/main/Emus/RPGMAKER/libs), [here](https://github.com/dokoma/kiwi/tree/main/Emus/LOVE/0.9.2/libs) and [here](https://github.com/dokoma/kiwi/tree/main/Emus/TEXT/libs) 😃 

I manage to run To The Moon with libs from this repo.

Don't knows if Portmaster can use shared libraries for all the ports...

**Comment by cizia64** [comment]
> Don't knows if Portmaster can use shared libraries for all the ports...

Probably not, I suppose that some ports requires specific lib versions. the order of the paths in LD_LIBRARY will be important (lib path from the port first, normally it shoul already be like that).

I plan to add a bunch of libs for PorsMaster in CrossMix v1.2.0, it will based on my ports tests and I'll include the libs from kiwi mentioned here too. 
I also work with PortMaster team to fix some stuffs like the controls which are often a problem with the TSP. 
These actions should improve PortMaster situation on for the TSP. 

However we still have a lack of 32bits libs and we will not have it soon (never ?). I had a conversation with TrimUI and the problem is that the GPU manufacturer didn't provide any 32bits libs for the PowerVR GE8300 GPU. There is a little hope for the future because Allwinner has also started to value this type of product but it will not happen very soon. So that means that all the 32bits ports will not work (box86 ports mostly I suppose),  +150 ports as of now. It will be necessary to inform the TSP community to avoid systematically questions about that.
